fvm/errors/errors.go

```diff
@@ -99,21 +106,29 @@ func HandleRuntimeError(err error) error {
 	return NewCadenceRuntimeError(runErr)
 }
 
-// TODO(patrick): combine ErrorCodeChecker, Error and Failure interface.
-// We can also combine CodedError and CodedFailure once the interface is
-// unified.
-type ErrorCodeChecker interface {
-	// TODO(patrick): add Code() ErrorCode
-	// TODO(patrick): add IsFailure() bool
+// This returns true if the error or one of its nested errors matches the
+// specified error code.
+func HasErrorCode(err error, code ErrorCode) bool {
+	return Find(err, code) != nil
+}
 
-	HasErrorCode(code ErrorCode) bool
+// This recursively unwraps the error and returns first CodedError that matches
+// the specified error code.
```

Comment:
@janezpodhostnik JFYI: we are now doing the recursive error search which may uncover bugs.